cmd/web-handlers.go

```diff
@@ -662,21 +661,20 @@ func (web *webAPIHandlers) Upload(w http.ResponseWriter, r *http.Request) {
 		pipeReader, pipeWriter := io.Pipe()
 		snappyWriter := snappy.NewWriter(pipeWriter)
 
-		actualReader, err := hash.NewReader(reader, size, "", "", actualSize)
+		var actualReader *hash.Reader
+		actualReader, err = hash.NewReader(reader, size, "", "", actualSize)
 		if err != nil {
 			writeWebErrorResponse(w, err)
 			return
 		}
+
 		go func() {
-			defer pipeWriter.Close()
-			defer snappyWriter.Close()
 			// Writing to the compressed writer.
-			_, err = io.CopyN(snappyWriter, actualReader, actualSize)
-			if err != nil {
-				hashError = err
-				return
-			}
+			_, cerr := io.CopyN(snappyWriter, actualReader, actualSize)
+			pipeWriter.CloseWithError(cerr)
+			snappyWriter.Close()
```

Comment:
The same here

Reply:
This I will change, we need to fix this in couple of other locations as well.